src/services/taskService.js

```diff
@@ -0,0 +1,302 @@
+//importing files...
+import api from "./axiosConfig";
+import { v4 as uuidv4 } from "uuid";
+
+
+//GET ENDPOINTS
+/**
+ * getAllTasks - Asynchronous function that gets all tasks in the db
+ * @param {object} queryParams: An object that contains the query parameters to retrieve tasks with
+ *                              or defaults to an empty object
+ * @returns {Promise<Array>} All tasks created
+ */
+export const getAllTasks = async (queryParams = {}) => {
+  try {
+    //Building the queryString...
+    const queryString = new URLSearchParams(queryParams).toString();
```

Comment:
Since we might not be able to filter by query using the "like" keyword and JSON-server doesnt filter date range; you might want to implement logic to manually filter those (if its included in the queryParams) after getting the result from JSON server.
Also while testing I realized if you included these two queries I mention; it makes the url invalid and returns an empty array.

So essentially your function should
1. Remove the keyword and date range parameters from the query Param first
2. Perform your request like you already are
3. Filter the keywork and date range parameters manually from the response.
4. return said response.

Do you understand?

Reply:
Got it.